docker/plugin/digitile-monstache-plugin.go

```diff
@@ -0,0 +1,459 @@
+		return nil, fmt.Errorf("Failed to lookup filetype %s for asset %s: %s",
+			filetypeId.Hex(), assetId, lookupError)
+	}
+
+	return
+}
+
+func LoadCategory(input *monstachemap.MapperPluginInput, filetypeId string,
+	categoryId primitive.ObjectID) (category bson.M, err error) {
+
+	client := input.MongoClient
+	database := input.Database
+
+	categoryFields := bson.D{
+		{"_id", 0},
+		{"name", 1},
+	}
+
+	lookupError := client.Database(database).Collection("file_category").
+		FindOne(context.Background(), bson.M{"_id": categoryId},
+			options.FindOne().SetProjection(categoryFields)).Decode(&category)
+
+	if lookupError != nil {
+		return nil, fmt.Errorf("Failed to lookup category %s for filetype %s: %s",
+			categoryId.Hex(), filetypeId, lookupError)
+	}
+
+	return
+}
+
+func AddTagNames(input *monstachemap.MapperPluginInput, document map[string]interface{}, wg *sync.WaitGroup,
+	documentFragments chan map[string]interface{}, lookupErrors chan string) {
+
+	defer wg.Done()
+
+	assetId := document["id"].(string)
+	tags := document["tags"].([]interface{})
+	tagIds := make([]primitive.ObjectID, len(tags))
+	documentFragment := make(map[string]interface{})
+
+	for index, tagId := range tags {
+		tagIds[index] = tagId.(primitive.ObjectID)
+	}
+
+	tagNames, err := GetTags(input, assetId, tagIds)
+
+	if err == nil {
+		documentFragment["tags"] = tagNames
+		documentFragments <- documentFragment
+	} else {
+		lookupErrors <- err.Error()
+	}
+}
+
+func GetTags(input *monstachemap.MapperPluginInput, assetId string, tagIds []primitive.ObjectID) (tags []string, err error) {
+
+	client := input.MongoClient
+	database := input.Database
+
+	tags = make([]string, len(tagIds))
+
+	tagsFilter := bson.M{
+		"_id": bson.M{
+			"$in": tagIds,
+		},
+	}
+
+	tagFields := bson.D{
+		{"_id", 1},
+		{"name", 1},
+	}
+
+	cur, lookupError := client.Database(database).Collection("tag").
+		Find(context.Background(), tagsFilter,
+			options.Find().SetProjection(tagFields))
+
+	if lookupError != nil {
+		return nil, fmt.Errorf("Failed to retrieve tags for asset %s: %s",
+			assetId, lookupError)
+	}
+
+	defer cur.Close(context.Background())
+
+	index := 0
+
+	tagIdsRetrieved := make([]string, 0, len(tagIds))
+
+	for cur.Next(context.Background()) {
+
+		tag := struct {
+			Id   string `bson:"_id"`
+			Name string `bson:"name"`
+		}{}
+
+		decodeErr := cur.Decode(&tag)
+
+		if decodeErr != nil {
+			return nil, fmt.Errorf("Failed to decode tag for asset %s: %s",
+				assetId, decodeErr)
+		}
+
+		tagIdsRetrieved = append(tagIdsRetrieved, tag.Id)
+
+		tags[index] = strings.ToLower(tag.Name)
+		index++
+	}
+
+	if cursorErr := cur.Err(); cursorErr != nil {
+		return nil, fmt.Errorf("Failed while iterating through tags for asset %s: %s",
+			assetId, cursorErr)
+	}
+
+	if len(tagIdsRetrieved) < len(tagIds) {
+		warnLog.Printf("Attempted to retrieve %d tags for asset %s but only retrieved %d with IDs %s",
+			len(tagIds), assetId, len(tagIdsRetrieved), tagIdsRetrieved)
+	}
```

Comment:
Only a warning is logged If the tags loaded do not match the tags referenced by the asset